library/alloc/src/boxed.rs

```diff
@@ -1361,6 +1369,29 @@ impl From<&str> for Box<str> {
 #[cfg(not(no_global_oom_handling))]
 #[stable(feature = "box_from_cow", since = "1.45.0")]
 impl From<Cow<'_, str>> for Box<str> {
+    /// Converts a `Cow<'_, str>` into a `Box<str>`
+    ///
+    /// When `cow` is the `Cow::Borrowed` variant, this
+    /// conversion allocates on the heap and performs a the
+    /// underlying `str`. Otherwise, it re-used the owned
+    /// string.
```

Comment:
Same as above.

```suggestion
    /// When `cow` is the `Cow::Borrowed` variant, this
    /// conversion allocates on the heap and performs a copy the
    /// underlying `str`. Otherwise, it try to re-use the owned
    /// string's allocation.
```

Reply:
```suggestion
    /// When `cow` is the `Cow::Borrowed` variant, this
    /// conversion allocates on the heap and performs a copy of the
    /// underlying `str`. Otherwise, it try to re-use the owned
    /// string's allocation.
```